### ISSUE TITLE
Bad operand question in backward training

### ISSUE THREAD

**Issue by isoflurane** [Ikarosy/Gated-LIF#3]
Hi, I'm trying train GLIF SNNs on static datasets, and the checkpoints-filename I used is "checkpoint_max_newest.pth", but encountered the problem with operands that did not conform to the python specification:
In layers.py, after saved_tensers in line 48, input should be of type tuple, whereas later hu calculations require input to be a number, such as int or float, which results in a bad operand error.
Looking forward to your answer, and thank you in advance.
<img width="571" alt="image" src="https://github.com/Ikarosy/Gated-LIF/assets/72650388/b8f071b3-c7d7-4b47-aa40-6463af83d420">

**Comment by Ikarosy** [comment]
Emm，I am not sure what error you encountered since you have changed my original codes obviously, and maybe show me the error report will help ? (not sure)

I guess you can try creating a virtual  environment with Anaconda and installing the PyTorch to my version as listed in the [Requirements](https://github.com/Ikarosy/Gated-LIF/tree/master#requirements).  Or, try using other surrogate gradient functions that would not trigger this error (according to my experiments, the surrogate gradient function would not affect the results pretty much).  And, GLIF has already been added to SpikingJelly.  You can just try using their latest SpikingJelly version, and set the neuron type to GLIF.  And, use their surrogate gradient implementation.

**Comment by Ikarosy** [comment]
The latest NeurIPS'24 paper has further verified the reproducibility and the performance of GLIF [Parallel Spiking Neurons with High Efficiency and Ability to Learn Long-term Dependencies ](https://arxiv.org/abs/2304.12760). You can also try contacting them and discuss about these implementation details of surrogate gradients.

**Comment by Ikarosy** [comment]
Re-open this issue if further discussion is requested
